### ISSUE TITLE
hardwareBackPress is not working in Webview

### ISSUE THREAD

**Issue by mhmtsrfglu** [react-native-community/react-native-webview#426]
I have a webview app. I want to active harwarebackpress on my webview. But my codes are not working.

`  webView = {
    canGoBack: false,
    ref: null,
  }

  onAndroidBackPress = () => {
    if (this.webView.canGoBack && this.webView.ref) {
      this.webView.ref.goBack();
      return true;
    }
    return false;
  }

  componentWillMount() {
    if (Platform.OS === 'android') {
      BackHandler.addEventListener('hardwareBackPress', this.onAndroidBackPress);
    }
  }

  componentWillUnmount() {
    if (Platform.OS === 'android') {
      BackHandler.removeEventListener('hardwareBackPress');
    }
  }`

`<WebView
            source={{ uri: website }}
            style={Platform.OS === 'ios'?styles.webviewsios:styles.webviewsandroid}
            automaticallyAdjustContentInsets={false}
            javaScriptEnabled={true}
            domStorageEnabled={true}
            decelerationRate="normal"
            startInLoadingState={true}
            useWebKit={true}
            ref={(webView) => { this.webView.ref = webView; }}
            onNavigationStateChange={(navState) => { this.webView.canGoBack = navState.canGoBack; }}
      
            />`

**Comment by Titozzz** [comment]
Need more details, nothing happens? It's crashing?

**Comment by winkelsdorf** [comment]
Works as expected here. As a note to @mhmtsrfglu: Some SPA websites mess up with the WebView history.

To narrow the problem down: Check with a breakpoint if your line `this.webView.ref.goBack();` is reached at all.

**Comment by mhmtsrfglu** [comment]
> Need more details, nothing happens? It's crashing?

nope. There is no error, or no crash. Even not log to console anything

**Comment by glacjay** [comment]
I'm using webview-10.8.3, and it will trigger `onNavigationStateChange` when go forward in webview, but don't trigger when go backward.
